sdk/eventhubs/azure_messaging_eventhubs/examples/eventhubs_tracing_subscriber.rs

```diff
@@ -0,0 +1,217 @@
+/// These constants are used to set properties in the EventData object.
+const LOG_LEVEL_PROPERTY: &str = "log_level";
+const NAME_PROPERTY: &str = "name";
+const TARGET_PROPERTY: &str = "target";
+const MODULE_PATH_PROPERTY: &str = "module_path";
+const FILE_PROPERTY: &str = "file";
+const LINE_PROPERTY: &str = "line";
+const FIELDS_PROPERTY: &str = "fields";
+const IS_SPAN_PROPERTY: &str = "is_span";
+const IS_EVENT_PROPERTY: &str = "is_event";
+const TIMESTAMP_PROPERTY: &str = "timestamp";
+
+impl<S> Layer<S> for EventHubsLayer
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+{
+    fn on_event(&self, event: &Event<'_>, _ctx: Context<'_, S>) {
+        // Format the event into a string
+        let mut visitor = EventVisitor {
+            buffer: String::new(),
+        };
+
+        event.record(&mut visitor);
+
+        // Add timestamp, level, and target
+        let metadata = event.metadata();
+
+        // Build an EventData object containing both the event body and properties reflecting the event's metadata.
+        let mut event_data_builder = EventData::builder()
+            .with_body(visitor.buffer)
+            .add_property(LOG_LEVEL_PROPERTY.to_string(), metadata.level().as_str())
+            .add_property(
+                TIMESTAMP_PROPERTY.to_string(),
+                SystemTime::now()
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .unwrap()
+                    .as_secs(),
+            )
+            .add_property(TARGET_PROPERTY.to_string(), metadata.target())
+            .add_property(NAME_PROPERTY.to_string(), metadata.name())
+            .add_property(FIELDS_PROPERTY.to_string(), metadata.fields().to_string())
+            .add_property(IS_EVENT_PROPERTY.to_string(), metadata.is_event())
+            .add_property(IS_SPAN_PROPERTY.to_string(), metadata.is_span());
+
+        // Handle optional metadata fields.
+        if let Some(module_path) = metadata.module_path() {
+            event_data_builder =
+                event_data_builder.add_property(MODULE_PATH_PROPERTY.to_string(), module_path);
+        }
+        if let Some(file) = metadata.file() {
+            event_data_builder = event_data_builder.add_property(FILE_PROPERTY.to_string(), file);
+        }
+        if let Some(line) = metadata.line() {
+            event_data_builder = event_data_builder.add_property(LINE_PROPERTY.to_string(), line);
+        }
+
+        let event_data = event_data_builder.build();
+
+        // Send the event to Event Hubs asynchronously
+        self.sender.try_send(event_data).unwrap_or_else(|_| {
+            eprintln!("Failed to send event to EventHubs");
+        });
+    }
+}
+
+/// A visitor that formats log messages and collects them into a buffer.
+struct EventVisitor {
+    buffer: String,
+}
+
+impl Visit for EventVisitor {
+    fn record_str(&mut self, field: &tracing::field::Field, value: &str) {
+        println!("record_str: {}: {}", field.name(), value);
```

Comment:
I feel like we shouldn't be printing to `stdout` in addition. If someone wants to "print" to EH and to `stdout`, let them add both layers. In that regard, we probably shouldn't even be printing errors to `stderr` as you are above, but I figured that was more of an error case where something didn't work right and this is a sample, so thought 🤷‍♂️.

Reply:
Those println's are primarily debugging. I'll pull them.